### PR TITLE
implement option validation function validateIsOption

### PR DESCRIPTION
**Pull request by rengelke**
* `validateIsOption()` to validate a list of options against specified criteria, improving robustness in option handling. This function ensures options match expected types, value ranges, allowed values, and handles `NULL` and `NA` values.

```r
options <- list(
optimizationMethod = "genetic_algorithm",
includeInteractions = TRUE,
maxIterations = 1000L,
convergenceThreshold = 0.02
)

validOptions <- list(
optimizationMethod = list(
type = "character", allowedValues = c("gradient_descent", "genetic_algorithm")
),
includeInteractions = list(type = "logical"),
maxIterations = list(type = "integer", valueRange = c(1L, 10000L)),
convergenceThreshold = list(type = "numeric", valueRange = c(0, 1))
)
```

* `validateColumns()` for rigorous validation of data frame columns against predefined specifications. Supports validation of type, value range, allowed values, and manages `NULL` and `NA` values effectively.

```r
df <- data.frame(
  age = c(25L, 30L, NA),
  BMI = c(22.5, 27.3, 24.9),
  gender = c("M", "F", "F"),
  smoker = c(TRUE, FALSE, NA)
)

columnSpecs <- list(
  age = list(type = "integer", valueRange = c(18L, 65L), naAllowed = TRUE),
  BMI = list(type = "numeric", valueRange = c(15.0, 40.0), naAllowed = TRUE),
  gender = list(type = "character", allowedValues = c("M", "F"), naAllowed = TRUE),
  smoker = list(type = "logical", naAllowed = TRUE)
)

validateColumns(df, columnSpecs)
```

* `validateVector()` introduced as a comprehensive vector validation tool, allowing checks against type, value range, and predefined allowed values with considerations for `NULL` and `NA`. Used by other validation functions `validateIsOption()` and `validateColumns()`.

```r
validateVector(x = 1:5, type = "integer")
validateVector(x = c(1.2, 2.5), type = "numeric", valueRange = c(1, 3))
validateVector(x = c("a", "b"), type = "character", allowedValues = c("a", "b", "c"))
validateVector(x = as.Date("2020-01-01"), type = "Date",
  valueRange = as.Date(c("2020-01-01", "2020-12-31")))
```

* `validateVectorRange()` and `validateVectorValues()` to provide detailed validation for value ranges and allowed values respectively. These functions complement `validateVector()`.

```r
# Range validation examples
validateVectorRange(x = c(5, 10), type = "numeric", valueRange = c(1, 10))
validateVectorRange(x = c("a", "b"), type = "character", valueRange = c("a", "c"))
validateVectorRange(x = as.Date(c("2020-01-01")), type = "Date",
  valueRange = as.Date(c("2020-01-01", "2020-12-31")))
validateVectorRange(x = 1:3, type = "integer", valueRange = c(1L, 5L))

# Allowed values validation examples
validateVectorValues(x = c("a", "b"), type = "character", allowedValues = c("a", "b", "c"))
validateVectorValues(x = c(2L, 4L), type = "integer", allowedValues = c(1L, 2L, 3L, 4L))
validateVectorValues(x = c(TRUE), type = "logical", allowedValues = c(TRUE, FALSE))
```

### CODE REVIEW

**Comment by PavelBal** [comment]
- @rengelke Please add a short description of the new function in the first comment (just copy-paste from the documentation and an example).

- Please add an entry to the `NEWS` file

**Comment by Felixmil** [comment]
Would this function could be very useful to validate a whole dataframe column (or even dataframe)? (i.e have vectors in inputOptions).

Also, did you take into account NULL or NAs ? Maybe allowing empty values could be an option in some cases ?

**Comment by Yuri05** [comment]
it seems that Ubuntu18 runner images are not supported by GitHub anymore (taht's why the 3 Ubuntu18 jobs are still pending)
https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories

**Comment by rengelke** [comment]
> this function could be very useful to validate a whole dataframe column (or even dataframe)?. Would this work (i.e have vectors in inputOptions) ?
> 
> Also, did you take into account NULL or NAs ? Maybe allowing empty values could be an option in some cases ?

Great idea about extending it to dataframes. It's totally doable with a few tweaks to make it more modular and flexible. We can definitely make it work for a column or dataframe.

About the NULL and NAs, I didn't focus much on them since we were validating options, where it wasn't so crucial. But including them makes a lot of sense for broader use.

I will get those changes done and commit..

**Comment by PavelBal** [comment]
@rengelke Whats the status of this one?

File: NEWS.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 NEW FUNCTIONS
 
+* `validateIsOption()` to validate input options against a list of valid options. It checks if the provided input options are valid based on a list of valid options, ensuring that each option matches the allowed types or values specified in `validOptions`. For options expecting numeric values, it checks if they fall within a specified range. For boolean options, it checks if they are strictly `TRUE` or `FALSE'.
```
Comment (PavelBal) [inline NEWS.md:5]: Also mention "validateVector" and "validateVectorRange"

**Review by Felixmil — APPROVED** [review]

**Pull request merged**